### DISCUSSION TITLE
Librespot and Archphile for Pi

### DISCUSSION THREAD

**Discussion by M-DF** [librespot-org/librespot#712] [General]
Hi,

I don't know if anyone can help with this issue.

I have a rasberry pi with an allo boss dac. I run archphile with bubbleupnp and qobuz. This works well. With the advent of cd quality spotify round the corner I thought I would experiment enabling librespot for Spotify connect. When I enable it to run at boot with 'systemctl enable librespot' spotify app sees the pi briefly and then it disappears. If I type librespot -n ''Archphile'' at the command prompt it works fine. This leads me to believe that there is something wrong in the librespot.service text

The librespot.service file text is as below;

[Unit]
Description=An open source client library for Spotify
After=network.target

[Service]
EnvironmentFile=/etc/librespot.conf
ExecStart=/usr/bin/librespot \
  --cache ${LIBRESPOT_CACHE} \
  --name Archphile \
  --device=plughw:0 \
  --bitrate ${LIBRESPOT_BITRATE} \
  --username ${LIBRESPOT_USERNAME} \
  --password ${LIBRESPOT_PASSWORD} \
  --initial-volume 100

Restart=always
#RestartSec=10
StartLimitInterval=30
StartLimitBurst=20
User=nobody
Group=audio

[Install]
WantedBy=multi-user.target

libresoft.conf file text is as below;

# Path to a directory where files will be cached
LIBRESPOT_CACHE=/tmp/librespot
# Bitrate (96, 160 or 320)
LIBRESPOT_BITRATE=320
# Username to sign in with
LIBRESPOT_USERNAME=
# Password
LIBRESPOT_PASSWORD=

Any thought would be welcome

**Comment by Johannesd3** [comment]
> With the advent of cd quality spotify round the corner I thought I would experiment enabling librespot for Spotify connect.

You shouldn't rely on that librespot supports it from the beginning.

> If I type librespot -n ''Archphile'' at the command prompt it works fine.

But the command in the service file seems to have a few more options, did you try them as well on the command line?

**Comment by roderickvd** [comment]
You might want to look into using `raspotify` that basically does this out of the box.
